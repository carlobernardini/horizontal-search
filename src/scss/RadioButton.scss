.orizzonte__radio--disabled {
    .orizzonte__radio-span--last {
        color: $text-color-muted;
    }
    .orizzonte__radio-label {
        cursor: not-allowed;

        &:hover {
            .orizzonte__radio-span--first {
                border-color: $border-color;
            }
        }
    }
}


.orizzonte__radio-input {
    display: none;

    &:checked {
        + .orizzonte__radio-label {
            .orizzonte__radio-span--first {
                background: $radio-color;
                border-color: $radio-color;
                animation: wave .4s ease;

                .orizzonte__radio-svg {
                    stroke-dashoffset: 0;
                }

                &:before {
                    transform: scale(2.5);
                    opacity: 0;
                    transition: all .4s ease;
                }
            }
        }
    }
}

.orizzonte__radio-label {
    margin: auto;
    user-select: none;
    cursor: pointer;

    &:hover {
        .orizzonte__radio-span--first {
            border-color: $color-b;
        }
    }
}

.orizzonte__radio-span {
    display: inline-block;
    vertical-align: middle;
    transform: translate3d(0,0,0);

    &--first {
        box-sizing: content-box;
        position: relative;
        width: 12px;
        height: 12px;
        border-radius: 50%;
        transform: scale(1);
        vertical-align: middle;
        border: 1px solid $border-color;
        transition: all .2s ease;

        &:before {
            content: "";
            width: 100%;
            height: 100%;
            background: $radio-color;
            display: block;
            transform: scale(0);
            opacity: 1;
            border-radius: 50%;
        }
    }

    &--last {
        padding: 5px 0 5px 7px;
        color: #000;
    }
    &--count {
        float: right;
        color: $text-color-muted;
        font-size: .9em;
        padding: 5px 0 5px 7px;
    }
}

.orizzonte__radio-svg {
    position: absolute;
    top: 0;
    left: 0;
    fill: #ffffff;
    transition: all .3s ease;
    transition-delay: .1s;
}
