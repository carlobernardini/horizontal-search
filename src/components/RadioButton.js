import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { DEFAULT_CHOICES_VIEWBOX } from '../constants';
import '../scss/RadioButton.scss';

const RadioButton = ({ disabled, id, label, name, onChange, value, viewBox, selected }) => (
    <div
        className={ classNames('orizzonte__radio', {
            'orizzonte__radio--disabled': disabled
        }) }
    >
        <input
            type="radio"
            disabled={ disabled }
            id={ id }
            name={ name }
            className="orizzonte__radio-input"
            value={ value }
            checked={ selected }
            onChange={ () => {
                onChange(value);
            }}
        />
        <label
            htmlFor={ id }
            className="orizzonte__radio-label"
        >
            <span
                className="orizzonte__radio-span orizzonte__radio-span--first"
            >
                <svg
                    className="orizzonte__radio-svg"
                    width="12px"
                    height="10px"
                    viewBox={ viewBox || DEFAULT_CHOICES_VIEWBOX }
                >
                    <circle
                        cx="5"
                        cy="6"
                        r="3"
                    />
                </svg>
            </span>
            <span
                className="orizzonte__radio-span orizzonte__radio-span--last"
            >
                { label }
            </span>
        </label>
    </div>
);

RadioButton.displayName = 'OrizzonteRadioButton';

RadioButton.propTypes = {
    /** If the radio button should be disabled */
    disabled: PropTypes.bool,
    /** Internal ID for this radio button */
    id: PropTypes.string.isRequired,
    /** Label for this radio button */
    label: PropTypes.string.isRequired,
    /** Name for current series of radio buttons */
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    /** If the radio button should be checked */
    selected: PropTypes.bool,
    /** Value for this radio button */
    value: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string
    ]).isRequired,
    viewBox: PropTypes.array
};

RadioButton.defaultProps = {
    disabled: false,
    onChange: () => {},
    selected: false,
    viewBox: null
};

export default RadioButton;
