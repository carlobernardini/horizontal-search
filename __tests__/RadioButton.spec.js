import React from 'react';
import RadioButton from '../src/components/RadioButton';

const onChange = jest.fn();

describe('<RadioButton />', () => {
    it('should render a radio button', () => {
        const value = 'test';
        const wrapper = shallow(
            <RadioButton
                id="testRadio1"
                name="testRadioGroup"
                label="Test radio"
                value={ value }
                onChange={ onChange }
            />
        );

        wrapper.find('.orizzonte__radio-input').simulate('change');
        expect(onChange).toHaveBeenCalledWith(value);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a disabled radio button', () => {
        const wrapper = shallow(
            <RadioButton
                id="testRadio1"
                name="testRadioGroup"
                label="Test radio"
                value="test"
                disabled
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a selected radio button', () => {
        const wrapper = shallow(
            <RadioButton
                id="testRadio1"
                name="testRadioGroup"
                label="Test radio"
                value="test"
                selected
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a radio button with facet count', () => {
        const wrapper = shallow(
            <RadioButton
                id="testRadio1"
                name="testRadioGroup"
                label="Test radio"
                value="test"
                facetCount={ 123 }
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
