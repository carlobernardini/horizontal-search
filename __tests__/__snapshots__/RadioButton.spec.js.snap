// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render a disabled radio button 1`] = `
<div
  className="orizzonte__radio orizzonte__radio--disabled"
>
  <input
    checked={false}
    className="orizzonte__radio-input"
    disabled={true}
    id="testRadio1"
    name="testRadioGroup"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <label
    className="orizzonte__radio-label"
    htmlFor="testRadio1"
  >
    <span
      className="orizzonte__radio-span orizzonte__radio-span--first"
    >
      <svg
        className="orizzonte__radio-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <circle
          cx="5"
          cy="6"
          r="3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__radio-span orizzonte__radio-span--last"
    >
      Test radio
    </span>
  </label>
</div>
`;

exports[`<RadioButton /> should render a radio button 1`] = `
<div
  className="orizzonte__radio"
>
  <input
    checked={false}
    className="orizzonte__radio-input"
    disabled={false}
    id="testRadio1"
    name="testRadioGroup"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <label
    className="orizzonte__radio-label"
    htmlFor="testRadio1"
  >
    <span
      className="orizzonte__radio-span orizzonte__radio-span--first"
    >
      <svg
        className="orizzonte__radio-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <circle
          cx="5"
          cy="6"
          r="3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__radio-span orizzonte__radio-span--last"
    >
      Test radio
    </span>
  </label>
</div>
`;

exports[`<RadioButton /> should render a radio button with facet count 1`] = `
<div
  className="orizzonte__radio"
>
  <input
    checked={false}
    className="orizzonte__radio-input"
    disabled={false}
    id="testRadio1"
    name="testRadioGroup"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <label
    className="orizzonte__radio-label"
    htmlFor="testRadio1"
  >
    <span
      className="orizzonte__radio-span orizzonte__radio-span--first"
    >
      <svg
        className="orizzonte__radio-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <circle
          cx="5"
          cy="6"
          r="3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__radio-span orizzonte__radio-span--last"
    >
      Test radio
    </span>
    <span
      className="orizzonte__radio-span orizzonte__radio-span--count"
    >
      123
    </span>
  </label>
</div>
`;

exports[`<RadioButton /> should render a selected radio button 1`] = `
<div
  className="orizzonte__radio"
>
  <input
    checked={true}
    className="orizzonte__radio-input"
    disabled={false}
    id="testRadio1"
    name="testRadioGroup"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <label
    className="orizzonte__radio-label"
    htmlFor="testRadio1"
  >
    <span
      className="orizzonte__radio-span orizzonte__radio-span--first"
    >
      <svg
        className="orizzonte__radio-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <circle
          cx="5"
          cy="6"
          r="3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__radio-span orizzonte__radio-span--last"
    >
      Test radio
    </span>
  </label>
</div>
`;
